### PR TITLE
feat: Improve type annotations in utils.py

### PR DESCRIPTION
**Pull request by bluetech**
Replace some Anys with proper types.

### CODE REVIEW

**Comment by untitaker** [comment]
Thanks for your continued efforts on this @bluetech!

**Comment by bluetech** [comment]
@untitaker This has one failure left:

```
sentry_sdk/utils.py:398: error: Argument 2 to "filename_for_module" has incompatible type "Optional[str]"; expected "str"
```

It looks legitimate but I'm not sure. The `module` variable in `serialize_frame` can be None, but `filename_for_module` doesn't except None. But `filename_for_module` is wrapped in a `try...except Exception` so it doesn't crash, but I'm not sure if that's intended.

Would you mind taking a look?

**Comment by untitaker** [comment]
Yeah, seems like i was a bit careless about the logic on that one

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -358,7 +358,10 @@ def safe_repr(value):
 
 
 def filename_for_module(module, abs_path):
-    # type: (str, str) -> str
+    # type: (str, Optional[str]) -> Optional[str]
```
Comment (bluetech) [inline sentry_sdk/utils.py]: Looking at `serialize_frame`, `module` can be `None` too.

Comment (untitaker) [reply sentry_sdk/utils.py]: You're right, not sure why mypy didn't catch that one. I think I fixed it now though!

**Comment by untitaker** [comment]
Thanks again!

**Pull request merged**